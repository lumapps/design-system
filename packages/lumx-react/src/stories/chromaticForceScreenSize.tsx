/* eslint-disable */
import React from 'react';
import isChromatic from 'chromatic/isChromatic';

/** Story decorator used to force a minimum screen size when running in chromatic */
export const chromaticForceScreenSize = (story: any) =>
    isChromatic() ? <div style={{ minWidth: 1200, minHeight: 800 }}>{story()}</div> : story();
