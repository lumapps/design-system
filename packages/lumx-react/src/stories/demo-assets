../../../site-demo/static/demo-assets