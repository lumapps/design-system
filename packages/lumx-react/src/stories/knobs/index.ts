export * from './image';
export * from './lorem';
