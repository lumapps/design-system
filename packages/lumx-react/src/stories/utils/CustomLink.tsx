/* eslint-disable jsx-a11y/anchor-has-content,jsx-a11y/anchor-is-valid */
import React from 'react';

/**
 * Example custom link that can be used in linkAs props.
 */
export const CustomLink: any = ({ ...props }) => <a href="#" data-custom-link {...props} />;
