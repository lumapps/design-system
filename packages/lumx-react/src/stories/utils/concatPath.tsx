export type PathPart = string | number;
export type Path = PathPart | Array<PathPart>;

/**
 * Concat flatten object path
 *
 * @example concatPath('foo', 'bar') // => 'foo.bar'
 * @example concatPath(['foo', 0]) // => 'foo[0]'
 * @example concatPath('foo', 0, ['bar']) // => 'foo[0].bar'
 */
export const concatPath = (...prefix: Path[]) => {
    const [first, ...rest] = prefix.flat();
    return rest.reduce<string>((acc, part) => {
        if (typeof part === 'number') return `${acc}[${part}]`;
        return `${acc}.${part}`;
    }, String(first));
};
