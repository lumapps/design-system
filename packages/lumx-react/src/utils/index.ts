/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
