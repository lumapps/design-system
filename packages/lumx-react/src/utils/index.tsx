export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
