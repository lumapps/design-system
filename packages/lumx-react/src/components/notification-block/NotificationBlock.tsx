import React, { ReactElement, Ref } from 'react';

import classNames from 'classnames';

import { Theme } from '@lumx/react';

import { COMPONENT_PREFIX } from '@lumx/react/constants';

import { GenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';

/**
 * Defines the props of the component.
 */
interface NotificationBlockProps extends GenericProps {
    /** After content element */
    after?: ReactElement;
    /** Before content element */
    before?: ReactElement;
    /** Date when the notification occurs */
    date?: string;
    /** Description message of the notification */
    description?: string;
    /** Title of the notification */
    title?: string;
    /** Reference passed to the wrapper. */
    notificationBlockRef?: Ref<HTMLDivElement>;
}

/**
 * Define the types of the default props.
 */
interface DefaultPropsType extends Partial<NotificationBlockProps> {}

/**
 * The display name of the component.
 */
const COMPONENT_NAME = `${COMPONENT_PREFIX}NotificationBlock`;

/**
 * The default class name and classes prefix for this component.
 */
const CLASSNAME: string = getRootClassName(COMPONENT_NAME);

/**
 * The default value of props.
 */
const DEFAULT_PROPS: DefaultPropsType = {
    theme: Theme.light,
};

/**
 * Render a notification in a block.
 *
 * @return The component.
 */
const NotificationBlock: React.FC<NotificationBlockProps> = ({
    after,
    before,
    date,
    description,
    title,
    className,
    notificationBlockRef,
    theme = DEFAULT_PROPS.theme,
    ...forwardedProps
}) => {
    return (
        <div
            {...forwardedProps}
            ref={notificationBlockRef}
            className={classNames(
                className,
                handleBasicClasses({
                    prefix: CLASSNAME,
                    theme,
                }),
            )}
        >
            {before && <div className={`${CLASSNAME}__before`}>{before}</div>}
            <div className={`${CLASSNAME}__content`}>
                <span className={`${CLASSNAME}__title`}>{title}</span>
                <p className={`${CLASSNAME}__description`}>{description}</p>
                <span className={`${CLASSNAME}__date`}>{date}</span>
            </div>

            {after && <div className={`${CLASSNAME}__after`}>{after}</div>}
        </div>
    );
};
NotificationBlock.displayName = COMPONENT_NAME;

export { CLASSNAME, DEFAULT_PROPS, NotificationBlock, NotificationBlockProps };
