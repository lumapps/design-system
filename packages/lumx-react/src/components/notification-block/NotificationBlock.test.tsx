import React, { ReactElement } from 'react';

import { mount, shallow } from 'enzyme';
import 'jest-enzyme';

import { CommonSetup, Wrapper, commonTestsSuite, expectStoriesToMatchSnapshots } from '@lumx/react/testing/utils';

import { CLASSNAME, NotificationBlock, NotificationBlockProps } from './NotificationBlock';
import * as stories from './NotificationBlock.stories';

/**
 * Defines what the `setup` function will return.
 */
interface Setup extends CommonSetup {
    props: Partial<NotificationBlockProps>;

    /**
     * The <div> element wrapper.
     */
    wrapper: Wrapper;
}

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 *
 * @param props  The props to use to override the default props of the component.
 * @param     [shallowRendering=true] Indicates if we want to do a shallow or a full rendering.
 * @return      An object with the props, the component wrapper and some shortcut to some element inside of the
 *                       component.
 */
const setup = (
    { ...propsOverrides }: Partial<NotificationBlockProps> = {},
    shallowRendering: boolean = true,
): Setup => {
    const props: NotificationBlockProps = {
        date: '2min ago',
        description: 'Description message',
        title: 'Title',
        ...propsOverrides,
    };

    const renderer: (el: ReactElement) => Wrapper = shallowRendering ? shallow : mount;

    const wrapper: Wrapper = renderer(<NotificationBlock {...props} />);

    return {
        props,
        wrapper,
    };
};

describe(`<${NotificationBlock.displayName}>`, () => {
    // 1. Test render via snapshot.
    describe('Snapshots and structure', () => {
        expectStoriesToMatchSnapshots(stories, NotificationBlock);
    });

    // Common tests suite.
    commonTestsSuite(setup, { className: 'wrapper', prop: 'wrapper' }, { className: CLASSNAME });
});
