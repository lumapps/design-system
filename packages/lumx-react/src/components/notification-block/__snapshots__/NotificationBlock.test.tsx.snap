// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationBlock> Snapshots and structure should render story avatar 1`] = `
<div
  className="lumx-notification-block lumx-notification-block--theme-light"
>
  <div
    className="lumx-notification-block__before"
  >
    <Avatar
      badge={
        <Badge
          color="red"
        >
          <Icon
            icon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
          />
        </Badge>
      }
      image="http://i.pravatar.cc/40"
      size="m"
    />
  </div>
  <div
    className="lumx-notification-block__content"
  >
    <span
      className="lumx-notification-block__title"
    >
      Title
    </span>
    <p
      className="lumx-notification-block__description"
    >
      Description message
    </p>
    <span
      className="lumx-notification-block__date"
    >
      2min ago
    </span>
  </div>
  <div
    className="lumx-notification-block__after"
  >
    <IconButton
      emphasis="low"
      icon="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
    />
  </div>
</div>
`;

exports[`<NotificationBlock> Snapshots and structure should render story icon 1`] = `
<div
  className="lumx-notification-block lumx-notification-block--theme-light"
>
  <div
    className="lumx-notification-block__before"
  >
    <Icon
      hasShape={true}
      icon="M20,2A2,2 0 0,1 22,4V16A2,2 0 0,1 20,18H6L2,22V4C2,2.89 2.9,2 4,2H20M4,4V17.17L5.17,16H20V4H4M6,7H18V9H6V7M6,11H15V13H6V11Z"
    />
  </div>
  <div
    className="lumx-notification-block__content"
  >
    <span
      className="lumx-notification-block__title"
    >
      Title
    </span>
    <p
      className="lumx-notification-block__description"
    >
      Description message
    </p>
    <span
      className="lumx-notification-block__date"
    >
      2min ago
    </span>
  </div>
  <div
    className="lumx-notification-block__after"
  >
    <IconButton
      emphasis="low"
      icon="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
    />
  </div>
</div>
`;
